### PR TITLE
Implemented simple resilience PDS storage read #1805

### LINKED ISSUE #1805: PDS: Handle job storage problems at job execution time
## Situation
Currently the PDS job will fail if the workspace setup phase has problems while fetching the JobStorage - even when this is only a temporarily problem

## Wanted
The PDS shall do this more resilient. Those changes shall be changeable at  job runtime (so an Admin can change this in PDS job executor configuration (or even SecHub later automatically - if there is a global setup for this)

## Solution
- `PDSWorkspaceService` must be changed, so when job storage problems appear, the fetching 
- To be able to change this at runtime, we introduce following new PDS job parameters :
   - `pds.config.jobstorage.read.resilience.retries.max'
   - `pds.config.jobstorage.read.resilience.retry.wait.seconds'
- Add those parameters into `PDSConfigurationAutoFix`, so there is a fallback behavior when not defined.

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #1805

### CODE REVIEW

File: sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/util/PDSResilientRetryExecutor.java [hunk 1]
```diff
@@ -29,67 +51,116 @@ public PDSResilientRetryExecutor(int maxRetries, ExceptionThrower<E> thrower, Cl
...
                 }
                 lastCause = e;
-                retries++;
             }
         }
 
         if (!done) {
-            thrower.throwException("Even after " + retries + " retries it was not possible to execute " + identifier + "! Will throw last cause.", lastCause);
+            LOG.warn("After {} attempts the execution of '{}' finally failed.", attempts, identifier);
+            thrower.throwException("Even after " + attempts + " attempts it was not possible to execute '" + identifier + "' ! Will throw last cause.",
```
Comment (Jeeppler) [inline sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/util/PDSResilientRetryExecutor.java:127]:
```suggestion
            thrower.throwException("Even after " + attempts + " attempts it was not possible to execute '" + identifier + "'! Will throw last cause.",
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Tests and logic look good 🥇 

I only found smaller issues. Have a look at the comments.

File: sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/util/PDSResilientRetryExecutor.java [hunk 2]
```diff
@@ -29,67 +51,116 @@ public PDSResilientRetryExecutor(int maxRetries, ExceptionThrower<E> thrower, Cl
...
+                retryNumber = attempts - 1;
+
+                if (retryNumber > 0) {
+                    waitBeforeNextRetry(identifier);
+                    LOG.info("Start retry {}/{} of '{}'", retryNumber, maxRetries, identifier);
                 }
                 result = callable.call();
                 done = true;
+                if (retryNumber > 0) {
+                    LOG.info("The execution of '{}' was succesful", retryNumber, identifier);
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/util/PDSResilientRetryExecutor.java:112]: - The retry number parameter must be removed
- identifier can be anything and shall give a hint whats going on here. The mechanism of using the identifier inside the log messages to explain the context is used everywhere.

I would remove the number but keep the identifier.

[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/util/PDSResilientRetryExecutor.java]: That makes sense.

**Review by Jeeppler — APPROVED** [review]
Looks good.

**Pull request merged**